### PR TITLE
Make the number of workers processing federated query configurable

### PR DESCRIPTION
**Pull request by SungJin1212**
This PR adds a `-tenant-federation.max-concurrent` flags to make the number of workers processing federated query configurable. and add a `cortex_querier_federated_tenants_per_query` histogram to track the number of tenants per query.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by yeya24** [comment]
Maybe I don't have the scenario of query federation. What's the usecase for this metric?

File: pkg/querier/tenantfederation/tenant_federation.go [hunk 1]
```diff
@@ -7,8 +7,11 @@ import (
 type Config struct {
 	// Enabled switches on support for multi tenant query federation
 	Enabled bool `yaml:"enabled"`
+	// MaxConcurrent The number of workers used for processing federated query.
+	MaxConcurrent int `yaml:"max_concurrent"`
 }
 
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.Enabled, "tenant-federation.enabled", false, "If enabled on all Cortex services, queries can be federated across multiple tenants. The tenant IDs involved need to be specified separated by a `|` character in the `X-Scope-OrgID` header (experimental).")
+	f.IntVar(&cfg.MaxConcurrent, "tenant-federation.max-concurrent", defaultMaxConcurrency, "The number of workers used for processing federated query.")
```
Comment (yeya24) [inline pkg/querier/tenantfederation/tenant_federation.go:16]: This is the number of workers for a single federated query, correct? The description is a bit misleading as it sounds like concurrency for shared for all federated queries

Comment (SungJin1212) [reply pkg/querier/tenantfederation/tenant_federation.go]: what about `The number of workers used to process each federated query.`?

**Review by yeya24 — APPROVED** [review]

**Review by friedrichg — APPROVED** [review]
thanks!

**Pull request merged**